src/core/ddsc/src/dds_topic.c

```diff
@@ -152,7 +152,11 @@ static dds_return_t dds_topic_delete (dds_entity *e)
   struct dds_ktopic * const ktp = tp->m_ktopic;
   assert (dds_entity_kind (e->m_parent) == DDS_KIND_PARTICIPANT);
   dds_participant * const pp = (dds_participant *) e->m_parent;
-  ddsi_sertopic_unref (tp->m_stopic);
+  type_identifier_t *type_id = ddsi_typeid_from_sertype (tp->m_stype);
```

Comment:
Yes, type identifiers can have variable size in the Xtypes spec, e.g. when using array or map collection types it will contain one or possibly two nested type identifiers. To be prepared for that, I've used dynamic allocation in the current implementation. The type hash field of the type identifier has a fixed length, currently 16 bytes and in the Xtypes implementation it will be 14 bytes.